driver-core/src/resources/META-INF/native-image/org.mongodb/mongodb-driver-core/native-image.properties

```diff
@@ -13,4 +13,4 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #
-Args = --initialize-at-run-time=com.mongodb.UnixServerAddress,com.mongodb.internal.connection.SnappyCompressor
+Args = --initialize-at-run-time=com.mongodb.UnixServerAddress,com.mongodb.internal.connection.SnappyCompressor,org.bson.types.ObjectId,com.mongodb.internal.connection.ClientMetadataHelper
```

Comment:
I'm a bit surprised this works because ObjectId is in the bson jar not driver-core, but I guess the native image compiler doesn't care.
But for anyone that is using the bson library directly, it seems like this more properly belongs in the bson jar file.

Reply:
I added ObjectId to bson. Tested by replacing mongodb-driver-sync with bson as a dependency, and confirming that the oid did not change between runs, and then changed after the fix.

Reply:
Looking at https://github.com/mongodb/mongo-java-driver/commit/9c3d62626fff5818ebb548fa72de5946dfccbada.  I'm surprised this works without the same change to `bson/build.gradle` as was made to `driver-core/build.gradle` in that commit. Any ideas, @rozza ?

Reply:
Regarding resources?  Seems we adopted in #527 a non-typical structure for resources in driver-core:  `src/resources` as opposed to the default location: `src/main/resources` https://docs.gradle.org/current/userguide/java_plugin.html

@katcharov used the default structure so no build changes needed.